File: backend/src/main/java/com/expense/tracker/model/SharedExpense.java
package com.expense.tracker.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.math.BigDecimal;

@Entity
@Table(name = "shared_expenses")
@Data
@NoArgsConstructor
public class SharedExpense {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "expense_id", nullable = false)
    private Expense expense;

    @ManyToOne
    @JoinColumn(name = "payer_user_id", nullable = false)
    private User payer;

    @ManyToOne
    @JoinColumn(name = "debtor_user_id", nullable = false)
    private User debtor;

    @Column(name = "amount_owed", nullable = false)
    private BigDecimal amountOwed;

    @Column(name = "is_settled")
    private boolean isSettled;
}

File: backend2/models/shared_expense.go
package models

type SharedExpense struct {
	ID           uint    `gorm:"primaryKey"`
	ExpenseID    uint    `gorm:"not null"`
	PayerUserID  uint    `gorm:"column:payer_user_id;not null"`
	DebtorUserID uint    `gorm:"column:debtor_user_id;not null"`
	AmountOwed   float64 `gorm:"column:amount_owed;not null;type:decimal(19,2)"`
	IsSettled    bool    `gorm:"column:is_settled"`

	Expense Expense `gorm:"foreignKey:ExpenseID"`
	Payer   User    `gorm:"foreignKey:PayerUserID"`
	Debtor  User    `gorm:"foreignKey:DebtorUserID"`
}
